compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -10918,6 +10818,8 @@ private STNode parseJoinClause(boolean isRhsExpr) {
      */
     private STNode parseOnClause(boolean isRhsExpr) {
         STNode onKeyword = parseOnKeyword();
+        // note that parsing expression includes following.
+        // equals-expr := expression `equals` expression
         STNode expression = parseExpression(OperatorPrecedence.QUERY, isRhsExpr, false);
```

Comment:
* `join-on-condition := on expression equals expression` from spec is parsed as `on-clause := on expression`.
i.e. we parse the `equals` keyword implicitly.  

* `equals` operator was added to error handler as well since it is already supported in the parser. 
There's no such operator in the spec though.